### ISSUE TITLE
openresty build error on Fedora 28.

### ISSUE THREAD

**Issue by tao12345666333** [openresty/openresty#367]
```
(Tao) ➜  ~ /usr/local/openresty/bin/openresty -V
nginx version: openresty/1.13.6.2
built by gcc 8.1.1 20180502 (Red Hat 8.1.1-1) (GCC) 
built with OpenSSL 1.0.2o-fips  27 Mar 2018
TLS SNI support enabled
configure arguments: --prefix=/usr/local/openresty/nginx --with-cc-opt=-O2 --add-module=../ngx_devel_kit-0.3.0 --add-module=../echo-nginx-module-0.61 --add-module=../xss-nginx-module-0.06 --add-module=../ngx_coolkit-0.2rc3 --add-module=../set-misc-nginx-module-0.32 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.08 --add-module=../srcache-nginx-module-0.31 --add-module=../ngx_lua-0.10.13 --add-module=../ngx_lua_upstream-0.07 --add-module=../headers-more-nginx-module-0.33 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.19 --add-module=../redis2-nginx-module-0.15 --add-module=../redis-nginx-module-0.3.7 --add-module=../rds-json-nginx-module-0.15 --add-module=../rds-csv-nginx-module-0.09 --add-module=../ngx_stream_lua-0.0.5 --with-ld-opt=-Wl,-rpath,/usr/local/openresty/luajit/lib --with-stream --with-stream_ssl_module --with-http_ssl_module
(Tao) ➜  ~ uname -a                             
Linux bogon 4.16.8-300.fc28.x86_64 #1 SMP Wed May 9 20:23:40 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

```

When I build openresty on Fedora 28, got some errors.

```
gmake[2]: 进入目录“/home/tao/tools/openresty-1.13.6.2/build/nginx-1.13.6”                                                  cc -c -I/home/tao/tools/openresty-1.13.6.2/build/luajit-root/usr/local/openresty/luajit/include/luajit-2.1 -I/home/tao/tools/openresty-1.13.6.2/build/luajit-root/usr/local/openresty/luajit/include/luajit-2.1  -pipe  -O -W -Wall -Wpointer-arith -W
no-unused-parameter -g -O2 -DNDK_SET_VAR -DNDK_UPSTREAM_LIST -DNDK_SET_VAR -DNDK_SET_VAR -DNDK_SET_VAR -DLUA_DEFAULT_PATH='
"/usr/local/openresty/site/lualib/?.ljbc;/usr/local/openresty/site/lualib/?/init.ljbc;/usr/local/openresty/lualib/?.ljbc;/u
sr/local/openresty/lualib/?/init.ljbc;/usr/local/openresty/site/lualib/?.lua;/usr/local/openresty/site/lualib/?/init.lua;/u
sr/local/openresty/lualib/?.lua;/usr/local/openresty/lualib/?/init.lua"' -DLUA_DEFAULT_CPATH='"/usr/local/openresty/site/lu
alib/?.so;/usr/local/openresty/lualib/?.so"' -DNDK_SET_VAR -I src/core -I src/event -I src/event/modules -I src/os/unix -I 
../ngx_devel_kit-0.3.0/objs -I objs/addon/ndk -I ../ngx_lua-0.10.13/src/api -I /home/tao/tools/openresty-1.13.6.2/build/lua
jit-root/usr/local/openresty/luajit/include/luajit-2.1 -I ../ngx_stream_lua-0.0.5/src/api -I objs \                        
        -o objs/src/os/unix/ngx_user.o \                                                                                   
        src/os/unix/ngx_user.c                                                                                             
src/os/unix/ngx_user.c: 在函数‘ngx_libc_crypt’中:                                                                          
src/os/unix/ngx_user.c:26:7: 错误：‘struct crypt_data’没有名为‘current_salt’的成员                                         
     cd.current_salt[0] = ~salt[0];                                                                                        
       ^                                                                                                                   
gmake[2]: *** [objs/Makefile:1197：objs/src/os/unix/ngx_user.o] 错误 1                                                     
gmake[2]: 离开目录“/home/tao/tools/openresty-1.13.6.2/build/nginx-1.13.6”                                                  
gmake[1]: *** [Makefile:8：build] 错误 2                                                                                   
gmake[1]: 离开目录“/home/tao/tools/openresty-1.13.6.2/build/nginx-1.13.6” 
```

This error is becauseof glibc bugs on 2.2.x and ngx add fix it at `bundle/nginx-1.13.6/src/os/unix/ngx_user.c +26 ` .

```
#ifdef __GLIBC__
    /* work around the glibc bug */
    cd.current_salt[0] = ~salt[0];
#endif
```

We can fix it by patch 

```
--- a/src/os/unix/ngx_user.c
+++ b/src/os/unix/ngx_user.c
@@ -21,8 +21,10 @@ ngx_libc_crypt(ngx_pool_t *pool, u_char *key, u_char *salt, u_char **encrypted)
     struct crypt_data   cd;
 
     cd.initialized = 0;
-#ifdef __GLIBC__
+#if (defined(__GLIBC__) && __GLIBC__ == 2) && (defined(__GLIBC_MINOR__) && \
+     __GLIBC_MINOR__ >= 2 && __GLIBC_MINOR__ < 4)
     cd.current_salt[0] = ~salt[0];
 #endif
 
```

**Comment by spacewander** [comment]
It is a known issue of Nginx: https://trac.nginx.org/nginx/ticket/1469

IMHO, we could wait for the upstream to fix it, since we have released the last version of OpenResty based on Nginx 1.13.

**Comment by agentzh** [comment]
Yeah, we should rely on the upstream fixes in either Fedora 28 or NGINX. I don't feel like hacking this thing ourselves in OpenResty.

**Comment by tao12345666333** [comment]
Yes. It has been fixed. https://github.com/nginx/nginx/commit/1ef5c9ce6f03907dad1677d25f108d2bf59a5c15

**Comment by tao12345666333** [comment]
The nginx 1.15.0 released.  https://github.com/nginx/nginx/releases/tag/release-1.15.0

**Comment by doc987** [comment]
Now that this is fixed, is there going to be a repository added for Fedora 28, or are there still outstanding issues?  The current repository list (https://openresty.org/package/fedora/) stops at Fedora 27.

**Comment by agentzh** [comment]
@doc987 If the issue has fixed in upstream Fedora 28, then we can provide an official OpenResty repo for that system. Thanks for the info.

**Comment by agentzh** [comment]
@doc987 Will you please confirm that OpenResty 1.13.6.2 can indeed get built successfully on the latest Fedora 28 system without changes?

**Comment by agentzh** [comment]
@tao12345666333 I believe nginx upstream has already removed that workaround.

**Comment by tao12345666333** [comment]
@agentzh  Yes, nginx upstream fixed it https://github.com/nginx/nginx/commit/1ef5c9ce6f03907dad1677d25f108d2bf59a5c15.  

And OpenResty 1.13.6.2 can't build successfully without changnes on Fedora 28. The nginx bundle is nginx-1.13.6, but the nginx upstream fixed this issue since 1.15.0.

**Comment by agentzh** [comment]
@tao12345666333 OK, then the next OpenResty formal release should have the fix since the next version will be based on NGINX 1.15.x.

**Comment by doc987** [comment]
Great!  Is there are schedule for the next OpenResty formal release, or is that unknown at this point?  It looks like the current formal release is 1.11.2.5 from August 30, 2017.

**Comment by agentzh** [comment]
@doc987 Where did you look? The latest version is 1.13.6.2 released on 14 May 2018:

https://openresty.org/en/download.html

The next OpenResty release will be in the next couple of months.

**Comment by doc987** [comment]
I was looking on the Github releases page (https://github.com/openresty/openresty/releases).  Release 1.13.6.1 (Nov 13, 2017) has the green "latest" release" label next to it.  The most recent release called a "formal release" is 1.11.2.5 (Aug 30, 2017).  1.13.6.2 is also shown as May 14, 2018, but without the "latest release" label (not sure why).  In any case 1.13.6.2 is definitely labeled as the latest release on the openresty.org download page you linked.

**Comment by thibaultcha** [comment]
This is fixed since OpenResty 1.15.8.1 is out (which includes the upstream Nginx patch for this).
